subsys/bluetooth/controller/Kconfig

```diff
@@ -372,6 +372,15 @@ config BT_CTLR_CONN_RSSI
 	help
 	  Enable connection RSSI measurement.
 
+config BT_CTLR_IGNORE_SAME_PEER_CONN
+	bool "Ignore connection requests from same peer"
+	default y
+	help
+	  Ignored connection requests from the same peer. While the
+	  Bluetooth specification does not allow multiple connections
+	  with the same peer, allowing such connections is useful
+	  while debugging multiple connections.
```

Comment:
If the spec specifies a `shall` for this, why is it a Kconfig? Wouldn't disabling this cause the stack to be non-compliant?

Reply:
This was suggested by @carlescufi https://github.com/zephyrproject-rtos/zephyr/issues/26172#issuecomment-644093770

Reply:
I see. Then I would suggest only allowing it to be turned on for TEST/DEBUG builds, e.g. by only allowing it be be `n` if e.g. `BT_TESTING` is enabled. 

I don't think it's a good idea to intentionally allowing a non-spec compliant build without some sort of warning or making sure that whoever disables this is aware of the consequences.

Reply:
I could make it dependent on `BT_CTLR_ADVANCED_FEATURES`, though I believe I would have to flip the polarity (`...ALLOW...` instead of `...IGNORE...` and `!defined` instead of `defined`) to get the correct behavior if the former is not defined.

Reply:
Alternatively we could add similar warnings to disabling this as we have when e.g. we enable SMP debug keys. See https://github.com/zephyrproject-rtos/zephyr/pull/26186/files

Reply:
I moved the user-controller flag it under `BT_CTLR_ADVANCED_FEATURES` and added a WARNING to the description. I also made the configuration now dependent on having more than one connection to not punish implementations that allow only a single connection anyway.